ethereum-common/src/network.rs

```diff
@@ -23,6 +23,17 @@ impl Network {
         .into()
     }
 
+    pub fn from_genesis_validators_root(root: &[u8]) -> Option<Self> {
+        let root_bytes: [u8; 32] = (*root).try_into().ok()?;
+        match root_bytes {
+            b if b == Self::Mainnet.genesis_validators_root().0 => Some(Mainnet),
+            b if b == Self::Sepolia.genesis_validators_root().0 => Some(Sepolia),
+            b if b == Self::Holesky.genesis_validators_root().0 => Some(Holesky),
+            b if b == Self::Hoodi.genesis_validators_root().0 => Some(Hoodi),
+            _ => None,
+        }
+    }
```

Comment:
for the sake of readability
```suggestion
    pub fn from_genesis_validators_root(root: [u8; 32]) -> Option<Self> {
        if root == Self::Mainnet.genesis_validators_root().0 {
             return Some(Mainnet);
        }
        
        if root == / * ... */
        
        None
    }
```